src/services/aws_macros.ts

```diff
@@ -81,6 +81,16 @@ export class AWS {
   constructor(config: AWSConfig) {
     // declare an specific slow retry strategy, to reuse in slow apis
     this.slowRetryStrategy = new StandardRetryStrategy(async () => SLOW_STRATEGY_RETRIES, {
+      retryDecider: error => {
+        // copied the default behavior from aws-sdk to fix the problem caused by Jest on isRetryableByTrait
+        if (!error) {
+          return false;
+        }
+        if (!error.message.includes('AWS SDK error wrapper'))
+          // default behavior when running outside Jest
+          return defaultRetryDecider(error);
+        else return true; // jest has messed the error object
+      },
```

Comment:
wouldn't be better to handle this using the `IASQL_ENV` variable to know if we are running with jest? I wonder if it is possible that any other error that does not include the `AWS SDK error wrapper` message should go through the `defaultRetryDecider` and we are changing that behaviour

Reply:
I agree with this. Furthermore, there are other retryDeciders in this file. Not just this one.